### PR TITLE
[ML] add support for question_answering NLP tasks

### PR DESCRIPTION
**Pull request by benwtrent**
Adds support for `question_answering` NLP models within the pytorch model uploader.

Related: https://github.com/elastic/elasticsearch/pull/85958

### CODE REVIEW

File: eland/ml/pytorch/transformers.py [hunk 1]
```diff
@@ -69,6 +71,86 @@
...
+            "position_ids": position_ids,
+        }
+
+        # remove inputs for specific model types
+        if isinstance(self._hf_model.config, transformers.DistilBertConfig):
+            del inputs["token_type_ids"]
+            del inputs["position_ids"]
+        response = self._hf_model(**inputs)
+        if isinstance(response, tuple):
+            return torch.stack(list(response), dim=0)
```
Comment (benwtrent) [inline eland/ml/pytorch/transformers.py:134]: our internal BERT inference logic takes the first element of the tuple if the returned value is a tuple. This does not work for question answering models as its a tuple of the starts and ends token scores.

**Review by sethmlarson — APPROVED** [review]

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by sethmlarson** [comment]
@benwtrent Looks like there's a merge conflict, can you fix this then I'll merge.

**Comment by benwtrent** [comment]
i got it @sethmlarson don't want to merge until the ES side is all approved and merged. Thanks!

**Pull request merged**